File: android/app/src/main/java/com/svgrendernativeformat/SvgImageViewManager.java
package com.svgrendernativeformat;

import androidx.annotation.NonNull;

import com.facebook.react.uimanager.SimpleViewManager;
import com.facebook.react.uimanager.ThemedReactContext;
import com.facebook.react.uimanager.annotations.ReactProp;

public class SvgImageViewManager extends SimpleViewManager<SvgImageView> {
    @NonNull
    @Override
    public String getName() {
        return "SvgImage";
    }

    @NonNull
    @Override
    protected SvgImageView createViewInstance(@NonNull ThemedReactContext themedReactContext) {
        return new SvgImageView(themedReactContext);
    }

    @ReactProp(name = "imageUri")
    public void setImageUri(SvgImageView view, String uri) {
        view.setImageUri(uri);
    }
}

File: android/app/src/main/java/com/svgrendernativeformat/SvgImageViewManager.kt
package com.svgrendernativeformat

import com.facebook.react.bridge.ReactContext
import com.facebook.react.bridge.ReadableMap
import com.facebook.react.uimanager.SimpleViewManager
import com.facebook.react.uimanager.ThemedReactContext
import com.facebook.react.uimanager.annotations.ReactProp
import com.svgrendernativeformat.svg.ReactSvgRenderer

class SvgImageViewManager(private val reactContext: ReactContext) :
    SimpleViewManager<SvgImageView>() {
    override fun getName(): String {
        return "SvgImageView"
    }

    override fun createViewInstance(themedReactContext: ThemedReactContext): SvgImageView {
        return SvgImageView(themedReactContext)
    }

    @ReactProp(name = "svgComponent")
    fun setSvgComponent(view: SvgImageView, props: ReadableMap) {
        val renderer = ReactSvgRenderer(view.context as ThemedReactContext, props)
        val bitmap = renderer.renderToImage()
        view.setImageBitmap(bitmap)
    }
}